Clarify sentence under narrowing -> discriminated-unions (#2245)
microsoft/TypeScript-Website: packages/documentation/copy/en/handbook-v2/Narrowing.md

```

Like with our first definition of `Shape`, this is still an error.

> When `radius` was optional, we got an error (only with [`strictNullChecks`](/tsconfig#strictNullChecks) enabled) because TypeScript couldn't tell whether the property was present.

Reviewer: ```suggestion
When `radius` was optional, we got an error (with [`strictNullChecks`](/tsconfig#strictNullChecks) enabled) because TypeScript couldn't tell whether the property was present.
```

Now that `Shape` is a union, TypeScript is telling us that `shape` might be a `Square`, and `Square`s don't have `radius` defined on them!

> Both interpretations are correct, but only the union encoding of `Shape` will cause an error when [`strictNullChecks`](/tsconfig#strictNullChecks) is disabled.

Reviewer: ```suggestion
Both interpretations are correct, but only the union encoding of `Shape` will cause an error regardless of how [`strictNullChecks`](/tsconfig#strictNullChecks) is configured.
```
